types/d3-array/index.d.ts

```diff
@@ -809,7 +819,114 @@ export function transpose<T>(matrix: ArrayLike<ArrayLike<T>>): T[][];
 export function zip<T>(...arrays: Array<ArrayLike<T>>): T[][];
 
 // --------------------------------------------------------------------------------------
-// Histogram
+// Iterables
+// --------------------------------------------------------------------------------------
+
+/**
+ * Returns true if the given test function returns true for every value in the given iterable.
+ * This method returns as soon as test returns a non-truthy value or all values are iterated over.
+ * Equivalent to array.every.
+ */
+export function every<T>(
+    iterable: Iterable<T>,
+    test: (value: T, index: number, iterable: Iterable<T>) => unknown
```

Comment:
Should the return type not be boolean for all test method?

Reply:
Not really. the return value will be checked for "truthy"
Look at https://github.com/microsoft/TypeScript/blob/master/lib/lib.es5.d.ts#L1763

Reply:
Yes, that's correct. I matched the types from the array methods in the TypeScript ES5 types.

Reply:
I saw it was expected to be truthy, but is there no better way to express it than `unknown`? I mean, boolean would be closer to reality (though admittedly, JavaScript is weird on this :D )

Reply:
No, there's no better way. The types for the array `every`, etc. are written by the TS team (I presume) and are obviously used quite extensively, so I think it makes sense to defer to what they came up with as far as the best way to express these types.

Reply:
I get your point, makes sense. I would still appreciate a less aggressive reply (at least that's how it feels), as we are all on the same boat and giving up our free time for this.

Reply:
My apologies, I in no way intended my reply to be aggressive.

Just so that I know for the future, in what did you find it aggressive? My goal was just to point out that I didn't come up with these types without precedent and that these types have been battle-tested since they're used by pretty much anyway who uses TypeScript, so I assume that they're the best that they can be. It was meant purely as a logical argument and I was not trying to imply any form of aggression.

I definitely appreciate the time you've given up to work on this and I absolutely agree that we're all on the same boat and should work together.

Reply:
No problem, written communication is not always easy to understand ;-). I guess your first sentence is what got me, like a very affirmative tone closing all possibilities of discussion. As I said, your argument is sensible, and I approved the PR. Let's close that chapter, I understand very well that your intent was in no way aggressive, and it might just have been the grumpy me reading a tone that was not there

Reply:
> I saw it was expected to be truthy, but is there no better way to express it than `unknown`? I mean, boolean would be closer to reality (though admittedly, JavaScript is weird on this :D )

If you would make it boolean TS would only allow you to return boolean.
You would then end up with a lot of `return !!foo`
`unknown` is actually the best way. The return value can be anything, as anything can be evaluated "truthy". We COULD return `any` as we don't use the value, but `unknown` is more semantically correct I think.

https://stackoverflow.com/questions/51439843/unknown-vs-any